rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -886,11 +886,7 @@ public List<ResolvedDependency> resolveDependencies(Scope scope, Map<GroupArtifa
                 // The dependency may be modified by the current pom's dependency management
                 d = getValues(d, depth);
                 try {
-                    if (d.getVersion() == null) {
-                        throw new MavenDownloadingException("No version provided", null, dd.getDependency().getGav());
-                    }
-
-                    if (d.getType() != null && (!"jar".equals(d.getType()) && !"pom".equals(d.getType()) && !"zip".equals(d.getType()))) {
+                    if (d.getVersion() == null || (d.getType() != null && (!"jar".equals(d.getType()) && !"pom".equals(d.getType()) && !"zip".equals(d.getType())))) {
```

Comment:
Do we have another way of leaving a warning for the user here?

Reply:
Not at this level I believe; we have that `MavenDownloadingException` and the try/catch wrapping `MavenDownloadingExceptions` to signify breaking issues; for non-breaking issues we sometimes add an informative marker that's not printed to the source, but that's not possible at this level. I think it's fine to tolerate the absence here and `continue`.